pkg/auth/providers/ldap/ldap_client.go

```diff
@@ -423,8 +433,33 @@ func (p *ldapProvider) searchLdap(query string, scope string, config *v3.LdapCon
 	}
 
 	for i := 0; i < len(results.Entries); i++ {
+		externalId := results.Entries[i].DN
 		entry := results.Entries[i]
-		principal, err := ldap.AttributesToPrincipal(entry.Attributes, results.Entries[i].DN, scope, p.providerName, config.UserObjectClass, config.UserNameAttribute, config.UserLoginAttribute, config.GroupObjectClass, config.GroupNameAttribute)
+
+		if p.samlSearchProvider() {
+			if strings.EqualFold("user", entityType) {
+				userLoginValues := ldap.GetAttributeValuesByName(entry.Attributes, config.UserLoginAttribute)
+				if len(userLoginValues) > 0 {
+					externalId = userLoginValues[0] // only support first
+				}
```

Comment:
so for user lookup to work properly, the user must ensure that the value of the userLoginAttribute field is the same as some value configured in the saml config, right?